src/org/rascalmpl/parser/uptr/recovery/LiteralMatcher.java

```diff
@@ -29,24 +29,14 @@ public LiteralMatcher(int[] literal) {
     }
 
     @Override
-    public MatchResult findMatch(int[] input, int startLocation, int maxLength) {
+    public int match(int[] input, int location) {
         int length = chars.length;
-        int limit = Math.min(startLocation + maxLength - length, input.length - length + 1);
-        for (int start=startLocation; start<limit; start++) {
-            boolean matches = true;
-            for (int i=0; i<length; i++) {
-                if (input[start + i] != chars[i]) {
-                    matches = false;
-                    break;
-                }
-            }
-
-            if (matches) {
-                return new MatchResult(start, length);
+        for (int i=0; i<length; i++) {
```

Comment:
same question here, how do we know that `input[location + i]` never reads past the input array?